From 0b27d23f899fe5e0a71d15e9a1d9bda9b4f4b9e7 Mon Sep 17 00:00:00 2001
From: mdeejay <mdjrussia@gmail.com>
Date: Mon, 3 Jul 2017 01:21:27 +0700
Subject: changes for MTK

Change-Id: I3ebe8d6d2734c5c574242f50c5b5606a5fc38570

diff --git a/server/NatController.cpp b/server/NatController.cpp
index cda8f5f..c966a11 100644
--- a/server/NatController.cpp
+++ b/server/NatController.cpp
@@ -376,7 +376,9 @@ int NatController::setForwardRules(bool add, const char *intIface, const char *e
         goto err_return;
     }
 
-    if (runCmd(ARRAY_SIZE(cmd4), cmd4) && add) {
+    // Revert "Don't start tethering if IPv6 RPF is not supported."
+    // STOPSHIP: Make this an error.
+    if (runCmd(ARRAY_SIZE(cmd4), cmd4) && add && false /* STOPSHIP */) {
         rc = -1;
         goto err_rpfilter;
     }
diff --git a/server/SoftapController.cpp b/server/SoftapController.cpp
index 1c59ec3..2e6d05c 100755
--- a/server/SoftapController.cpp
+++ b/server/SoftapController.cpp
@@ -54,7 +54,8 @@ using android::base::WriteStringToFile;
 #endif
 
-std:: string hostapd_unix_file(StringPrintf("/data/misc/wifi/hostapd/wlan0"));
+//std:: string hostapd_unix_file(StringPrintf("/data/misc/wifi/hostapd/wlan0"));
+  std:: string hostapd_unix_file(StringPrintf("/data/misc/wifi/hostapd/ap0"));
 #ifdef LIBWPA_CLIENT_EXISTS
 static const char HOSTAPD_DHCP_DIR[]    = "/data/misc/dhcp";
 #endif
 static const char HOSTAPD_CONF_FILE[]    = "/data/misc/wifi/hostapd.conf";
@@ -289,7 +290,8 @@ int SoftapController::setSoftap(int argc, char *argv[]) {
             "hw_mode=%c\n"
             "ignore_broadcast_ssid=%d\n"
             "wowlan_triggers=any\n",
-            argv[2], argv[3], channel, (channel <= 14) ? 'g' : 'a', hidden));
+        //  argv[2], argv[3], channel, (channel <= 14) ? 'g' : 'a', hidden));
+            "ap0", argv[3], channel, (channel <= 14) ? 'g' : 'a', hidden));
 
     std::string fbuf;
     if (argc > 7) {
