From 6a6427dcee24664d42b448dcef4f9cb35f4e0433 Mon Sep 17 00:00:00 2001
From: mdeejay <mdjrussia@gmail.com>
Date: Mon, 3 Jul 2017 01:13:24 +0700
Subject: changes for MTK

Change-Id: I85ad7a3acccdd64f0e80edba7b7824801407219a

diff --git a/core/jni/fd_utils-inl.h b/core/jni/fd_utils-inl.h
index a68a60274ac..c6fc5088577 100644
--- a/core/jni/fd_utils-inl.h
+++ b/core/jni/fd_utils-inl.h
@@ -58,6 +58,7 @@ static const char* kPathWhitelist[] = {
   "/dev/urandom",
   "/dev/ion",
   "/dev/dri/renderD129", // Fixes b/31172436
+  "/proc/ged", // MTK add
   "/system/framework/org.cyanogenmod.platform-res.apk",
 #ifdef PATH_WHITELIST_EXTRA_H
 PATH_WHITELIST_EXTRA_H
